tutorials/2d/2d_lights_and_shadows.rst

```diff
@@ -0,0 +1,226 @@
+location of the light source. A child :ref:`Sprite <class_Sprite>` is not necessary to make a 
+:ref:`Light <class_Light2D>` work but is used to show the location of the light.
+
+.. image:: img/light_shadow_light_blob.png
+
+Shadows
+-------
+
+Shadows are made by intersecting a :ref:`Light <class_Light2D>` with a :ref:`LightOccluder2D <class_LightOccluder2D>`.
+
+By default shadows are turned off. To turn them on click on the :ref:`Light <class_Light2D>` 
+and under the Shadows section check ``Enabled``.
+
+In the demo we are using a :ref:`Sprite <class_Sprite>` with a Texture on it to make the "Shadow Casters", 
+but in reality all you need is a couple :ref:`LightOccluder2Ds <class_LightOccluder2D>`. By itself 
+the :ref:`LightOccluder2D <class_LightOccluder2D>` looks like a dark spot and in this demo the :ref:`Sprite <class_Sprite>` is 
+just a black square.
+
+Step by step
+------------
+
+Now that we have covered the basics of the nodes that will be used, we will now walk step by step through 
+the process of making a scene like the one found in the demo.
+
+First add a :ref:`Sprite <class_Sprite>` and set its texture to the `background image <https://raw.githubusercontent.com/godotengine/godot-demo-projects/master/2d/lights_and_shadows/bg.png>`_. For your game this can be whatever 
+your background will be. For this style of shadows it is most likely to be a floor texture. 
+
+.. image:: img/light_shadow_background.png
+
+Next create three :ref:`Light2D's <class_Light2D>` and set their textures to the `light image <https://raw.githubusercontent.com/godotengine/godot-demo-projects/master/2d/lights_and_shadows/light.png>`_. You can alter their 
+color in the top section. By default shadows are turned off and the ``mode`` is set to ``add``. This 
+means that each light adds its own color to whatever is underneath.
+
+.. image:: img/light_shadow_all_lights_no_blob.png
+
+Next child a :ref:`Sprite <class_Sprite>` to each of the :ref:`Light <class_Light2D>` nodes, and set 
+the :ref:`Sprite's <class_Sprite>` texture to the `blob image <https://raw.githubusercontent.com/godotengine/godot-demo-projects/master/2d/lights_and_shadows/spot.png>`_. Each of these 
+should stay centered on the :ref:`Light <class_Light2D>` node. The blob is the image of the light 
+itself while the :ref:`Light <class_Light2D>` shows the effect that the light has on the scene. The 
+:ref:`LightOccluder2D's <class_LightOccluder2D>` will treat the position of the light as the center of the :ref:`Light <class_Light2D>` 
+node which is why we want the blob to be centered on its parent :ref:`Light <class_Light2D>`.
+
+.. image:: img/light_shadow_all_lights.png
+
+.. note:: At the time of writing this tutorial the stable version is 3.0 but the master version that will become 3.1
+          contains many changes to the animation system. So the animations contained in the demo will not be covered.
```

Comment:
At the time of writing, 3.0 is the stable release version. The 3.1 development branch contains many changes to the animation system, so the animations in the demo will not be covered here. See :ref:`Introduction to 2D Animation. <doc_introduction_2d>` for more information.